docs/references/making_release.md

```diff
@@ -1,16 +1,16 @@
 ## How to make a new release on PyPI and conda-forge
 
 ### PyPI
-1. iterate the version number. Our version numbering approach is:
+1. update the changelog to put all the unreleased changes under the new version (leaving the unreleased section empty).
+2. Make a PR for the version change and get it accepted & merged. Only make a PyPI distribution from a clean master branch (no local changes).
+3. iterate the version number as a new tag. Our version numbering approach is:
```

Comment:
we can probably add something like `git tag <hash> vX.Y.Z` to show how to do it explicitly. Also mention that the hash should correspond exactly to the merge commit.